### PR TITLE
[SPARK-LLAP-207] Added HiveWarehouseSession API and related helper classes

### PR DESCRIPTION
**Pull request by EricWohlstadter**
## What changes were proposed in this pull request?

Added HiveWarehouseSession API and related helper classes

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
@EricWohlstadter . You can create issues here, https://github.com/hortonworks-spark/spark-llap/issues, and use ID as `SPARK-LLAP-XXX`. That helps us a lot later during backporting.

**Comment by jdere** [comment]
New files are missing apache headers.

**Comment by dongjoon-hyun** [comment]
Nice catch, @jdere !
@EricWohlstadter . I merged all the pending PRs and updated Travis CI command to use our internal repo temporarily inside this specific branch. Next time, Travis CI build will pass. Could you make another PR for adding Apache License Header?

File: src/main/java/com/hortonworks/spark/sql/hive/llap/HWConf.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.hortonworks.spark.sql.hive.llap;
+
+import java.util.Optional;
+
+enum HWConf {
+
+  USER("user.name", warehouseKey("user.name"), ""),
+  PASSWORD("password", warehouseKey("password"), ""),
+  HS2_URL("hs2.url", warehouseKey("hs2.url"), "jdbc:hive2://localhost:10084"),
```
Comment (jdere) [inline src/main/java/com/hortonworks/spark/sql/hive/llap/HWConf.java]: Looks like you were probably using this for testing, but a better default port might be 10500 (default port for HS2 with LLAP)

Comment (EricWohlstadter) [reply src/main/java/com/hortonworks/spark/sql/hive/llap/HWConf.java]: Addressed in https://github.com/hortonworks-spark/spark-llap/pull/211

**Comment by EricWohlstadter** [comment]
Added missing file headers in https://github.com/hortonworks-spark/spark-llap/pull/211

File: src/main/java/com/hortonworks/spark/sql/hive/llap/HiveWarehouseSession.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+public interface HiveWarehouseSession {
+
+    String HIVE_WAREHOUSE_CONNECTOR = "com.hortonworks.spark.sql.hive.llap.HiveWarehouseConnector";
+
+    Dataset<Row> executeQuery(String sql);
+    Dataset<Row> q(String sql);
+
+    Dataset<Row> execute(String sql);
+    Dataset<Row> exec(String sql);
```
Comment (HyukjinKwon) [inline src/main/java/com/hortonworks/spark/sql/hive/llap/HiveWarehouseSession.java]: @EricWohlstadter, sorry for leaving a comment late like this but `exec` is a reserved keyword in Python. If you don't mind will just take this out in Python. Seems a short name for `execute`. Python side uses a trailing underscore, for example, `exec_` in this case but I think it's fine to not add it if it's just a short name for `execute`.

Comment (dongjoon-hyun) [reply src/main/java/com/hortonworks/spark/sql/hive/llap/HiveWarehouseSession.java]: +1. Could you open an issue for that, @HyukjinKwon ?

Comment (EricWohlstadter) [reply src/main/java/com/hortonworks/spark/sql/hive/llap/HiveWarehouseSession.java]: @HyukjinKwon 

Ok, let's just take out "exec" all together. Please open an issue and I'll make the change for the Java side.

**Pull request merged**